### ISSUE TITLE
The focus in the table is often unable to be input. Occasionally, the focus is lost. I don't know where the mouse focus has gone

### ISSUE THREAD

**Issue by UniKylin** [marktext/marktext#3513]
### Versions

- MarkText version: v0.16.3
- Operating system: MacOS Monterey 12.6

**Comment by cha0ran** [comment]
Yes, this problem still exists in the updated version, non-English input loses focus and content insertion can only be done in code mode

**Comment by Jocs** [comment]
Thanks for the report (and for the follow-up confirming the IME symptom). This dates back to v0.16.3, which used MarkText's old table implementation where a whole table shared a single editable region — that's what caused the caret to jump/disappear and non-English (IME) input to fail in table cells.

MarkText's editor engine has since been rewritten. In the current engine each table cell is its own editable block, and there's dedicated handling for IME composition inside cells (including the WebKit/macOS case where typing CJK into an empty cell used to corrupt the table). This is covered by unit and e2e tests. Given that, I'm closing this as resolved by the engine migration.

If you still hit lost focus or IME input problems on a build running the new engine, please reopen with your MarkText version and the exact steps (OS, input method, and whether the cell was empty) and we'll take another look. Thanks!
